File: grading/3-benchmarks+codegen/valid/_insertionsort.py
size = 25000
arr = [0] * size

def insertion_sort():
    global size
    global arr
    for j in xrange(1, size):
        key = arr[j]
        i = j - 1
        while i >= 0 and arr[i] > key:
            arr[i+1] = arr[i]
            i = i - 1
        arr[i+1] = key

def main():
    global size
    global arr
    j = size - 1
    while j >= 1:
        arr[size - j - 1] = j
        j = j - 1
    insertion_sort()
    for i in range(10):
        print arr[i]

if __name__ == "__main__":
    main()

File: grading/3-benchmarks+codegen/valid/insertionsort.go
//~0
//~1
//~2
//~3
//~4
//~5
//~6
//~7
//~8
//~9
//&Python: 54.6 seconds
//&C++: 7.4 seconds

package main

var size = 25000
var arr [25000]int

func insertionSort() {
	for j := 0; j < size; j++ {
		key := arr[j]
		i := j - 1
		for i >= 0 && arr[i] > key {
			arr[i+1] = arr[i]
			i = i - 1
		}

		arr[i+1] = key
	}
}

func main() {
	for j := size - 1; j >= 0; j-- {
		arr[size - j - 1] = j
	}
	insertionSort()
	for i := 0; i < 10; i++ {
		println(arr[i])
	}

}
